engine/src/test/java/io/zeebe/engine/state/WorkflowExecutionCleanZeebeStateTest.java

```diff
@@ -0,0 +1,618 @@
+    workflowInstanceSubscriptionState = zeebeState.getWorkflowInstanceSubscriptionState();
+    messageStartEventSubscriptionState = zeebeState.getMessageStartEventSubscriptionState();
+
+    assertThatStateIsEmpty();
+  }
+
+  @Test
+  public void testWorkflowWithServiceTask() {
+    engineRule
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .serviceTask("task", t -> t.zeebeJobType("test"))
+                .endEvent()
+                .done())
+        .deploy();
+
+    final var workflowInstanceKey =
+        engineRule.workflowInstance().ofBpmnProcessId(PROCESS_ID).withVariable("x", 1).create();
+
+    engineRule
+        .job()
+        .ofInstance(workflowInstanceKey)
+        .withType("test")
+        .withVariable("y", 2)
+        .complete();
+
+    RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_COMPLETED)
+        .withWorkflowInstanceKey(workflowInstanceKey)
+        .withElementType(BpmnElementType.PROCESS)
+        .await();
+
+    assertThatStateIsEmpty();
+  }
+
+  @Test
+  public void testWorkflowWithSubprocess() {
+    engineRule
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .subProcess(
+                    "subprocess",
+                    subProcess ->
+                        subProcess
+                            .zeebeInputExpression("x", "y")
+                            .zeebeOutputExpression("y", "z")
+                            .embeddedSubProcess()
+                            .startEvent()
+                            .endEvent())
+                .endEvent()
+                .done())
+        .deploy();
+
+    final var workflowInstanceKey =
+        engineRule.workflowInstance().ofBpmnProcessId(PROCESS_ID).withVariable("x", 1).create();
+
+    RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_COMPLETED)
+        .withWorkflowInstanceKey(workflowInstanceKey)
+        .withElementType(BpmnElementType.PROCESS)
+        .await();
+
+    assertThatStateIsEmpty();
+  }
+
+  @Test
+  public void testWorkflowWithMultiInstance() {
+    engineRule
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .serviceTask(
+                    "task",
+                    t ->
+                        t.zeebeJobType("test")
+                            .multiInstance(
+                                m ->
+                                    m.zeebeInputCollectionExpression("items")
+                                        .zeebeInputElement("item")
+                                        .zeebeOutputCollection("results")
+                                        .zeebeOutputElementExpression("result")))
+                .endEvent()
+                .done())
+        .deploy();
+
```

Comment:
no given, when then comments anymore ?

Reply:
Nope, but I have no strong option in this case :sweat_smile:  

For you, what is the value of this comments here?

Reply:
It makes it more clear and is easier to read if you have the separators you can clearly see the sections. For me it is often that I can easier/faster understand the test.

Reply:
:+1: